src/role-access/containers/allocate-role/choose-role/choose-role.component.spec.ts

```diff
@@ -1,63 +1,103 @@
-import { MockStore, provideMockStore } from '@ngrx/store/testing';
-import { State } from '../../../../app/store/reducers';
+import { of } from 'rxjs';
+
 import { ChooseRadioOptionComponent } from '../../../components';
 import { CHOOSE_A_ROLE } from '../../../constants';
+import { AllocateRoleNavigationEvent, AllocateRoleStateData } from '../../../models';
+import { RoleAssignmentService } from '../../../services';
 import { ChooseRoleComponent } from './choose-role.component';
 
+const firstRoles = [{ roleId: '1', roleName: 'Role 1' },
+      { roleId: '2', roleName: 'Role 2' },
+      { roleId: '3', roleName: 'Role 3' }];
+
+const firstRoleOptions = [{ optionId: '1', optionValue: 'Role 1' },
+      { optionId: '2', optionValue: 'Role 2' },
+      { optionId: '3', optionValue: 'Role 3' }];
+
+const secondRoles = [{ roleId: '1', roleName: 'Role 1' },
+      { roleId: '2', roleName: 'Role 2' },
+      { roleId: '3', roleName: 'Role 3' }];
+
+const mockAllocateRoleStateData: AllocateRoleStateData = {
+  caseId: '1234',
+  state: null,
+  typeOfRole: null,
+  allocateTo: null,
+  person: null,
+  durationOfRole: null,
+  period: null
+};
+
 describe('ChooseRoleComponent', () => {
   const radioOptionControl: FormControl = new FormControl('');
   const formGroup: FormGroup = new FormGroup({[CHOOSE_A_ROLE]: radioOptionControl});
 
   let component: ChooseRoleComponent;
   let fixture: ComponentFixture<ChooseRoleComponent>;
-  let store: MockStore<State>;
+  const mockStore = jasmine.createSpyObj('store', ['dispatch', 'pipe']);
 
-  let spyOnPipeToStore = jasmine.createSpy();
-  let spyOnStoreDispatch = jasmine.createSpy();
+  const roleAssignmentService = {
+    judicialRoles: firstRoles,
+    legalOpsRoles: secondRoles
+  };
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       schemas: [CUSTOM_ELEMENTS_SCHEMA],
       declarations: [ ChooseRadioOptionComponent, ChooseRoleComponent ],
       imports: [
+        HttpClientModule,
         ReactiveFormsModule
       ],
       providers: [
-        provideMockStore(),
+        { provide: Store, useValue: mockStore },
         {
           provide: ActivatedRoute,
           useValue: {
             snapshot: {
               queryParams: {
-                userJourney: 'judicial'
+                userType: 'judicial'
               }
             },
           }
         },
+        { provide: RoleAssignmentService, useValue: roleAssignmentService }
       ]
     })
     .compileComponents();
   }));
 
   beforeEach(() => {
-    store = TestBed.get(Store);
-    spyOnPipeToStore = spyOn(store, 'pipe').and.callThrough();
-    spyOnStoreDispatch = spyOn(store, 'dispatch');
     fixture = TestBed.createComponent(ChooseRoleComponent);
     component = fixture.componentInstance;
     component.formGroup = formGroup;
+    mockStore.pipe.and.returnValue(of(mockAllocateRoleStateData));
     fixture.detectChanges();
   });
 
   it('should create', () => {
```

Comment:
Might as well remove this test now that you have added other tests.

Reply:
Done